### PR TITLE
Mes 8438 date picker styling

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
replaced instances of ion-datetime buttons with a new component that is more aesthetically fitting
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image
image

### CODE REVIEW

File: src/app/pages/office/components/date-of-test/date-of-test.ts [hunk 1]
```diff
@@ -91,4 +93,20 @@ export class DateOfTest implements OnInit {
 
   disableEdit = () => this.editMode = false;
 
+  handleEvents(dateTime: IonDatetime, buttonType: string) {
```
Comment (andrewsetterfield) [inline src/app/pages/office/components/date-of-test/date-of-test.ts]: can this have a return type added

**Comment by andrewsetterfield** [comment]
Can you move this to the common components section and make use of it from there.  dont like that we're importing the entire test module into office to gain access to a single component.

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts [hunk 2]
```diff
@@ -164,4 +164,17 @@ export class AdvancedSearchComponent {
         },
       ).finally(() => dateTime.confirm(true));
   };
+
+  changeDate(event: { control?: string; data: string }) {
```
Comment (andrewsetterfield) [inline src/app/pages/test-results-search/components/advanced-search/advanced-search.ts]: can we add a return type please

File: src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    </ion-input>
+    <ion-modal
+            class="{{displayType}}-modal"
+            trigger="dateTimeItem + {{control}}"
+            [keepContentsMounted]="true"
+            mode="ios"
+            [backdropDismiss]="false">
+        <ng-template>
+            <ion-datetime
+                    *ngIf="displayType === 'date'"
```
Comment (andrewsetterfield) [inline src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html]: dont like the string comparison here.  can we use an enum?

File: src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                    (ngModelChange)="onSelected(time, displayType)">
+                <ion-buttons slot="buttons" *ngIf="emitButtonEvents">
+                    <ion-button color="primary" (click)="time.cancel(); buttonEmit(time, 'cancel')">Cancel</ion-button>
+                    <ion-button *ngIf="showClearButton" color="primary" (click)="time.reset(); buttonEmit(time, 'clear')">Clear</ion-button>
+                    <ion-button color="primary" (click)="time.confirm(); buttonEmit(time, 'done')">Done</ion-button>
+                </ion-buttons>
+            </ion-datetime>
+        </ng-template>
+    </ion-modal>
+</div>
```
Comment (andrewsetterfield) [inline src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html]: EOF

File: src/app/pages/test-results-search/components/datetime-input/date-time-input.component.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+.input-box {
+  border-radius: 6px;
+  border: 2px solid #222222;
+  box-shadow: inset 0 3px 3px #a5a2a2;
+  color: black;
+  width: 150px;
+  text-align: center;
+}
```
Comment (andrewsetterfield) [inline src/app/pages/test-results-search/components/datetime-input/date-time-input.component.scss]: EOF

File: src/app/pages/office/components/date-of-test/date-of-test.ts (hunk 1 above)
Comment (RLCorp) [reply src/app/pages/office/components/date-of-test/date-of-test.ts]: addressed in new commit

File: src/app/pages/test-results-search/components/advanced-search/advanced-search.ts (hunk 2 above)
Comment (RLCorp) [reply src/app/pages/test-results-search/components/advanced-search/advanced-search.ts]: addressed in new commit

File: src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html (hunk 3 above)
Comment (RLCorp) [reply src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html]: addressed in new commit

File: src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html (hunk 4 above)
Comment (RLCorp) [reply src/app/pages/test-results-search/components/datetime-input/date-time-input.component.html]: addressed in new commit

File: src/app/pages/test-results-search/components/datetime-input/date-time-input.component.scss (hunk 5 above)
Comment (RLCorp) [reply src/app/pages/test-results-search/components/datetime-input/date-time-input.component.scss]: addressed in new commit

**Review by andrewsetterfield — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Pull request merged**